### ISSUE TITLE
Request answers ，please，Thank you

### ISSUE THREAD

**Issue by 8cyh** [abubakar-sani/anti_jamming#2]
Dear author, hello. I have installed the ns3gym package and there should be no issues with environment setup. However, this error persists during runtime, which is Got new port for ns3g interface: 8626, ns3 file not found Quitting. Can you help me with the answer? Thank you very much.

**Comment by abubakar-sani** [comment]
@8cyh could you please attach a screenshot of the error

**Comment by 8cyh** [comment]
Hello author, this is the screenshot of the error, anti Jamming_ V1 and AntiJamming_ V2, V3, and V4 all run with this result. Do you know the reason and solution? Thank you very much for your answer。




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Abubakar Sani ***@***.***&gt;; 
发送时间: 2023年8月14日(星期一) 下午3:22
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [abubakar-sani/anti_jamming] Request answers ，please，Thank you (Issue #2)





 
@8cyh could you please attach a screenshot of the error

**Comment by 8cyh** [comment]
![微信图片_20230814152708](https://github.com/abubakar-sani/anti_jamming/assets/141317954/246bd7a4-6813-491f-9a8a-0341aefbab67)
May I ask if the author can see the picture？

**Comment by abubakar-sani** [comment]
@8cyh I can see that you have installed the ns3gym package. but you also need to install ns3 package first. Use this link to install ns3 https://www.nsnam.org/wiki/Installation

**Comment by 8cyh** [comment]
Dear author, I have already installed the ns3 installation package the last time, but it still ran with this result. May I ask what is the reason for this? Is it because of the version or something? Thank you very much for your answer

**Comment by abubakar-sani** [comment]
@8cyh It seems like you're encountering an issue related to the ns3gym package during runtime. The error message "Got new port for ns3g interface: 8626, ns3 file not found Quitting." suggests that the package is trying to locate a specific NS-3 simulation file but is unable to find it. Here are a few steps you can take to troubleshoot and resolve the issue:

1. Check NS-3 Installation: Make sure that you have NS-3 properly installed on your system and that the necessary environment variables are set correctly. The ns3gym package relies on NS-3 for simulation, so any issues with your NS-3 installation could potentially cause this error.

2. Path Configuration: Ensure that the path to the NS-3 installation is correctly set in your environment variables. The ns3gym package might be looking for the NS-3 files in a specific directory, and if the path is incorrect, it won't be able to locate the required files.

**Comment by 8cyh** [comment]
Thank you for your answer. Let me give it a try. Finally, may I ask if there is a requirement for Ubuntu or NS-3 version in your code? Because your introduction did not mention the version of Ubuntu, thank you.

**Comment by abubakar-sani** [comment]
@8cyh I used ubuntu version 20.04

**Comment by 8cyh** [comment]
May I ask the author what version of NS-3 do you use? 3.26 or 3.27 or 3.28? I am using Ubuntu 22.04, is this okay? Thank you for your answer. Wishing you a happy life.

**Comment by 8cyh** [comment]
Hello author, excuse me. I have already installed ns3gym, but according to what you said, I haven't installed ns-3 yet. Do I have to uninstall ns3gym first, and then reinstall ns3gym after installing ns-3? Thank you very much

**Comment by abubakar-sani** [comment]
@8cyh yes you should uninstall ns3gym then install ns3 and ns3gym again.

**Comment by 8cyh** [comment]
Dear author, hello. In your introduction to GitHub code requirements, the ns3 version requires 3.30, and the ns3gym version requires 1.0.2. However, the 1.0.2 version of ns3gym on the official website is compatible with the 3.36 version of ns3. Is this okay?Thank you for your answer！！wish you best

**Comment by abubakar-sani** [comment]
@8cyh Sorry for the delayed response. Kindly use the updated versions to run your codes

**Comment by Qiu0421** [comment]
Dear author, hello. I have installed all the packages and there should be no issues with environment setup. However, this error persists during runtime, which is in main raise Exception("Couldn't find the specified program: %s" % target_to_run)
Exception: Couldn't find the specified program: anti_jamming-main.
Can you help me with the answer? Thank you very much.
